internal/httpserve/handlers/webhook.go

```diff
@@ -0,0 +1,89 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+	"github.com/stripe/stripe-go/v81/webhook"
+	echo "github.com/theopenlane/echox"
+
+	ent "github.com/theopenlane/core/internal/ent/generated"
+)
+
+// WebhookReceiverHandler handles incoming stripe webhook events
+func (h *Handler) WebhookReceiverHandler(ctx echo.Context) error {
+	req := ctx.Request()
+	res := ctx.Response()
+
+	const MaxBodyBytes = int64(65536)
+	req.Body = http.MaxBytesReader(res.Writer, req.Body, MaxBodyBytes)
+
+	payload, err := io.ReadAll(req.Body)
+	if err != nil {
+		return ctx.String(http.StatusServiceUnavailable, fmt.Errorf("problem with request. Error: %w", err).Error())
+	}
+
+	event, err := webhook.ConstructEvent(payload, req.Header.Get("Stripe-Signature"), h.Entitlements.Config.StripeWebhookSecret)
+	if err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Errorf("error verifying webhook signature. Error: %w", err).Error())
+	}
+
+	exists, err := h.checkForEventID(ctx.Request().Context(), event.ID)
+	if err != nil {
+		return h.InternalServerError(ctx, err)
+	}
+
+	if exists {
+		log.Warn().Msgf("Event already received: %v", event.ID)
+
+		out := WebhookResponse{
+			Message: "Event already received",
+		}
+
+		return h.Success(ctx, out)
+	}
+
+	if !exists {
+		log.Warn().Msgf("Event not exists loop: %v", event.ID)
+
+		if err := h.Entitlements.HandleEvent(ctx.Request().Context(), &event); err != nil {
+			return h.InternalServerError(ctx, err)
+		}
+
+		input := ent.CreateEventInput{
+			EventID:   &event.ID,
+			EventType: "stripe",
+			// TODO unmarshall event data into internal event
+		}
+
+		meowevent, err := h.createEvent(ctx.Request().Context(), input)
+		if err != nil {
+			return h.InternalServerError(ctx, err)
+		}
+
+		log.Warn().Msgf("Internal event: %v", meowevent)
+	}
```

Comment:
this if isn't needed; if the `exists == true` you have a return, so everything would fall into this if 

Also don't think you want these warning logs? 

```suggestion
        if err := h.Entitlements.HandleEvent(req.Context(), &event); err != nil {
	        return h.InternalServerError(ctx, err)
        }
        
        input := ent.CreateEventInput{
	        EventID:   &event.ID,
	        EventType: "stripe",
	        // TODO unmarshall event data into internal event
        }
        
        meowevent, err := h.createEvent(ctx.Request().Context(), input)
        if err != nil {
	        return h.InternalServerError(ctx, err)
        }        
```